### ISSUE TITLE
Problems after MAX_FRAME REACHED

### ISSUE THREAD

**Issue by SuxRumpf** [Clothooo/lvt2calib#10]
Hello,
after 2 or 3 positions the software prints  MAX_FRAME REACHED and simply stops. There is no error or warning message.  Unfortunately is the result after 2 or 3 positions not precise enough. Is this a bug or am I doing something wrong here.
I am using the ubuntu 20.04/noetic version.

**Comment by Clothooo** [comment]
Hello, woudl you mind giving me more info or screenshots about it?

**Comment by SuxRumpf** [comment]
ok sry my post was misleading. 
I tried a couple things and recorded rosbags. It seems that the program always stops with specific rosbags.
It doesn't really breaks. the 2 terminal windows still show that circles in the image and  the board itself in the cloud is still found, but nothing happens in the main window. In my cases it always happens after MAX_FRAME of rgb reached.
With the rosbag it always breaks at  " [livox_horizon: x/60 rgb: 60/60] ".
I didn't look in your code so I don't have any idea what can cause this
![Screenshot from 2024-03-05 10-44-07](https://github.com/Clothooo/lvt2calib/assets/46651381/a969de02-6a6b-40c9-86e3-86239482a55b)

I am using Ubuntu 20.04 and therefore ros noetic with your latest noetic version.
Furthermore i am using a livox horizon and a rgb cam.

I hope this helps

**Comment by Clothooo** [comment]
From your screenshot, the 'laser_pattern', 'camera_pattern', and 'pattern_collection' were still working properly. 'pattern_collection' will always wait for the frame number of the features from livox reaching 60. In this picture, it was only 13, far from 60.
Sorry, I'm not sure what your 'break' means. Could you give me more details?

**Comment by SuxRumpf** [comment]
the pattern_collection stops. there is no error or warning it just stops. i could send the rosbag maybe you can reproduce the error.
It freezes and never reaches the query asking the user if they want to continue.

**Comment by Clothooo** [comment]
Yeah, it would be better if you can share the rosbag with us. In additon, only the 'pattern_collection' freezes? How about the 'laser_pattern'?

**Comment by SuxRumpf** [comment]
ok where should i submit you the bag ?
i cannot submit the bag here. its to big

**Comment by Clothooo** [comment]
Could you use Google Drive or OneDrive?

**Comment by SuxRumpf** [comment]
this is the [rosbag](https://fhaachen-my.sharepoint.com/:u:/g/personal/mr8631s_ad_fh-aachen_de/ESX34W5McVVMgn4aMDreVgUBBmGoHqsvFZqx-EKgsRSAmQ?e=Ngq5xL) in OneDrive

**Comment by Clothooo** [comment]
Hello, thank you for your sharing. I tested your data and your issue didn't occur on my platform. Our algorithm is not very fast, so it will take some time to wait for the 'pattern_collection’. As long as the 'laser_pattern' doesn't stop, our algorithm is still trying to calculate and run.
(Because the camera intrinsic wasn't provided, I just ran using your lidar data as shown in this figure.)
![image](https://github.com/Clothooo/lvt2calib/assets/31653202/6c895c68-700b-4389-a07e-43eea6c356f0)
